lib/api/src/backend/sys/entities/function/mod.rs

```diff
@@ -300,16 +318,16 @@ impl Function {
             let args_sig = args_sig.clone();
             let results_sig = results_sig.clone();
             let func = func.clone();
-            let args = match typed_args_from_values::<Args>(raw_store, args_sig.as_ref(), values) {
+            let args =
+                match typed_args_from_values::<Args>(raw_store, args_sig.as_ref(), values) {
                 Ok(args) => args,
                 Err(err) => return Box::pin(async { Err(err) }),
             };
-            let future = (*func)(env_mut, args);
+            let future = func
+                .as_ref()
+                .call_async(AsyncFunctionEnv::with_env(env_mut), args);
```

Comment:
I don't think this is good. It enables calling an async function with call instead of call_async. 
It also implicitly creates a Coroutine under the hood by calling `call_async`.